### DISCUSSION TITLE
Optional Message for Coin Purchases/Sales

### DISCUSSION THREAD

**Discussion by AikonMishima** [deso-protocol/dips#46] [Medium CIPs]
cip: To be assigned
title: Optional Message for Coin Purchases/Sales
author: Aikon (@aikonmishima)
discussions-url: N/A
created: 2021-07-12
One-Line Description
**An optional message on the "buy" and "sell" screen of a Creator Coin, and feature to motivate both the purchase or sale of a Creator Coin.**

----------------------------------------------------

### Explanation and Motivation
The purchases & sales of a Creator Coin are currently seen as simple transactions. The introduction of an optional message would be of importance for those who want to show their appreciation for their fans, or for the Creators to understand why people are selling off their Creator Coin (feedback).

![optional message bitclout](https://user-images.githubusercontent.com/84419699/125299038-08754700-e329-11eb-8405-c296d7768218.png)

### Goals and Non Goals
This CIP will not oblige anyone to send a message when they're buying or selling a Creator Coin. Sending a message is fully optional, and should remain optional.
The goal is to bring Creators and their Fans closer, by letting the Fans send a message to their favorite creator, while showing their appreciate through the purchase of the Creator Coin.

### Specification
On the "buy" / "sell" page, you will see an optional message (cfr image above).
When filled in, the Creator who got bought/sold will see this message in their notifications, only.

This CIP is in reference to discussion #15 , where the messaging variable could be repurposed for carrying the message payload.

Other methods can be used.
My purpose in opening this discussion is to bring forth ideas of implementing this in the smoothest & most scalable manner.

### Strengths
This CIP will help build relationships on the platform to a deeper level.
It will give Creators more insight as to why Fans buy their coin, and why they sold.
It will give Fans the ability to show why they love their favorite Creators' content, or why they don't like it anymore.

### Weakness
As with all channels of communication, intent & language are not always pure & courteous.

Negative messages could be written in the "sell" optional message, leading to discontent and demotivation of Creators.

### Backwards Compatibility
No issues.

### Test Cases
Can be done in a sandboxed node.

### Security Considerations
No additional security requirements.

### Data Increase
This CIP will add a small amount of data to a number (not all) of buy / sell transactions.

### Alternatives considered
None at the moment, as the current work around is to screenshot the buy / sell and submit a post, tagging the Creator.

### Acknowledgements
Thanks @Bloated-Devish for proposing to split the CIP's.

**Comment by AikonMishima** [comment]
Edited the initial post to clean up spelling/grammar mistakes and to make the layout more readable.

**Comment by bloated-devish** [comment]
@AikonMishima Thank you for doing this! Overall LGTM

## Few implementation questions
1. Will this message stay in the ```ExtraData``` associated with the buy/sell? (OR) Will this be a separate post that you can link? 

1a. If this is ```ExtraData``` associated with the transaction, please specify the limit of text length for this transaction. 
1b. If this is separate post - are there race conditions we need to worry - where the post isn't materialized by the network, but the sell is and this causes a situation where the linked post may not exist in the chain when there is a bug etc? Will we guarantee that the post will also be eventually be part of the chain.
  If the post is separate post, will it show up in the profile page of the seller/buyer? This is a product question, with arguments both sides:
  a. Show post - allows people to see that the account supports this creator.
  b. Don't show post - may be the message while public is addressed to the creator - so just show it to the creator? 
 Either ways, it should show up in the notification.

2. Can you also make a screenshot on how it will look in notification - so its clear to implementor what changes need to be done in frontend?

**Reply by bloated-devish** [reply]
@tijno This was my question to @AikonMishima on this issue. ```ExtraData``` will make it available technically to be used elsewhere, but as described on another CIP on soft-fork guidance, I am wondering whether dumping so many fields like this on ```ExtraData```would abuse the map.

**Comment by matt-mcmarsh** [comment]
a message on CCoin/DeSo sends would be good - possibly links with this(?)
